lib/gcp/token_validator.go

```diff
@@ -120,8 +121,15 @@ func (id *IDTokenValidator) Validate(ctx context.Context, token string) (*IDToke
 	return &claims, nil
 }
 
-// gcpServiceAccountEmailRegex is the regex used to extract the project ID from
-// the email claim of a GCP service account.
-// See https://cloud.google.com/iam/docs/service-accounts#service_account_email_addresses
-// See https://cloud.google.com/iam/docs/service-account-types#user-managed
-var gcpServiceAccountEmailRegex = regexp.MustCompile(`(?s)^[^@]+@([^\.]+)\.iam\.gserviceaccount\.com$`)
+var (
+	// gcpManagedServiceAccountEmailRegex is the regex used to extract the project ID from
```

Comment:
Nit: I think without the word "user" this is potentially more confusing - these are SAs created and managed by users of GCP, not SAs managed by GCP (which is what the Default Compute SA is)